src/reusable-components/SearchBar.tsx

```diff
@@ -1,28 +1,31 @@
-import { useRef } from "react";
+import { useState } from "react";
 
 type Props = {
   placeholder: string,
   onSearchTextChanged: Function
 }
 
 export default function SearchBar({ placeholder, onSearchTextChanged }: Props) {
-  const searchBarRef = useRef<HTMLInputElement>(null);
+  const [searchValue, setSearchValue] = useState<string>('')
 
   const updateSearchText = (searchText: string) => {
     onSearchTextChanged(searchText);
+    setSearchValue(searchText)
   }
 
   const clearSearchText = () => {
     onSearchTextChanged("");
-    if (searchBarRef.current) {
-      searchBarRef.current.value = "";
-    }
+    setSearchValue('')
   }
 
   return (
     <>
-      <input ref={searchBarRef} type="text" maxLength={20} placeholder={placeholder} onChange={(e) => updateSearchText(e.target.value)} className="rounded-xl mx-3 mt-6 pl-3 pr-5 py-1 border border-transparent focus:outline-none focus:border-accent dark:bg-black" />
-      {searchBarRef?.current?.value && <button onClick={() => clearSearchText()} className="h-3 w-3 absolute top-6 right-5">✗</button>}
+      {/*//SG if you want to pass the event value to a function you can add the function without using an arrow function onChange{updateSearchText}*/}
+      <input value={searchValue} type="text" maxLength={20}
+        onChange={(e) => updateSearchText(e.target.value)}
+        placeholder={placeholder} className="rounded-xl mx-3 mt-6 pl-3 pr-5 py-1 border border-transparent focus:outline-none focus:border-accent dark:bg-black" />
+      {/*SG You don't need an arrow function to invoke functions !!Watch out don't add () to invoke functions if you use it this way because it will fire at page load*/}
```

Comment:
It's ok to add () to functions if you write it like {() => function()}, it won't trigger at page load. It only triggers if you do not add "() =>" before such as "onChange = {function()}" which will fire at page load indeed. But yes, it's not necessary to write "{() => function()}" if you don't pass any parameter, so I updated this one as I missed it.